src/hardware/link-mso19/api.c

```diff
@@ -267,7 +276,10 @@ static int config_set(uint32_t key, GVariant *data,
 		trigger_pos = (int)pos;
 		devc->trigger_holdoff[0] = trigger_pos & 0xff;
 		break;
-	case SR_CONF_RLE:
+	case SR_CONF_COUPLING:
+		if (!cg_is_analog(cg) || (idx = std_str_idx(data, ARRAY_AND_SIZE(coupling))) < 0)
```

Comment:
a lot of stuff going on in this conditional, suggest splitting assignment and conditional on 2 lines